client/src/helper-functions/rpHelpers.js

```diff
@@ -36,55 +37,61 @@ const helper = {
         'photoUrl': firstResult['photos'][0],
       };
     })
-    // console.log(Object.values(allPropsObj, '✅'))
-
-
-    //console.log(`allPropsObj: ${JSON.stringify(Object.values(allPropsObj))}`)
     return Object.values(allPropsObj);
   },
 
   compileYourOutfitDataToProps: (currentProductInfo, currentProductStyles) => {
-
-     //console.log(`currentProductInfo: ${JSON.stringify(currentProductInfo)}`)
-     //console.log(`currentProductStyles: ${JSON.stringify(currentProductStyles)}`)
-    //  console.log(currentProductStyles)
-// console.log(currentProductInfo)
     let outfitPropsObj = {};
     let currentProductInfoCopy = Object.assign(currentProductInfo);
-    // console.log('hi')
 
     outfitPropsObj['product_id'] = currentProductInfoCopy.id;
     outfitPropsObj['name'] = currentProductInfoCopy.name;
     outfitPropsObj['category'] = currentProductInfoCopy.category;
-    //console.log(`🤠  beforedefaultResult: ${JSON.stringify(outfitPropsObj)}`)
-    // console.log(currentProductInfo, currentProductStyles)
-    const defaultResult = currentProductStyles.results.filter(result =>
 
-    result['default?'] === true)[0] ?? currentProductStyles.results[0];
+    const defaultResult = currentProductStyles.results.filter(result => result['default?'] === true)[0] ?? currentProductStyles.results[0];
 
-    //console.log(`defaultResult: ${JSON.stringify(defaultResult)}`);
     outfitPropsObj['originalPrice'] = defaultResult.original_price;
     outfitPropsObj['salePrice'] = defaultResult.sale_price;
     outfitPropsObj['photoUrl'] = defaultResult['photos'][0];
 
-
-    //console.log(`🐮 afterdefaultResult: ${JSON.stringify(outfitPropsObj)}`);
-    //console.log(`🤠 outfitPropsObj: ${JSON.stringify(outfitPropsObj)}`)
-
     return outfitPropsObj;
-
   },
 
   addIdToStylesData: (stylesData, productId) => {
-    //console.log(`stlesData: ${JSON.stringify(stylesData)}`)
-    //console.log(`productId: ${productId}`)
     let stateStylesCopy = Object.assign(stylesData);
     let productStylesWithId = {};
     productStylesWithId['product_id'] = productId;
     productStylesWithId['results'] = stateStylesCopy;
     return productStylesWithId;
-  }
+  },
 
+  formatFeatures: (currentProd, clickedProd) => {
```

Comment:
new function looks good - probably could have better time complexity.... just like all of our sorting functions, but it seems like this was one of the missing pieces for you.  good job 🤙

Reply:
Yea, I definitely do a lot of looping. The consolation is that there are never going to be more than maybe 4 features for each product total, so it would never have to handle things larger than that.